### DISCUSSION TITLE
Tensorflow & Opencv

### DISCUSSION THREAD

**Discussion by salarizadi** [Moddable-OpenSource/moddable#1091] [Ideas]
Hello to all.

It would be great if we could add opencv and tensorflow to moddable. I am researching this, trying to figure out how to do it, but with your help we will add more opencv, tensorflow and AI to moddable. Let me know if you have any ideas.

**Comment by phoddie** [comment]
What are you thinking to use as a starting point?

**Reply by salarizadi** [reply]
I'm on a trip right now, I'll be sure to check back after the trip and let you know the starting point, however we need to integrate a small version of these two systems with xs.

**Reply by salarizadi** [reply]
Since we need to process the image quickly, the module should be written for the ESP32 camera. I hope it has already been written?? Because in this way everything will go quickly or to start the XS simulator should be ready to work with the camera

**Reply by phoddie** [reply]
We have done some work on ESP32 camera a while back. It is unpublished because it felt very unstable. We could make it work with very specific configurations but nothing generally useful -- everything requires lots of tweaking. Maybe it has gotten better since then. Do you have any recent experience with ESP32 camera?

**Reply by salarizadi** [reply]
I will check this too, the esp32 camera is very important, we are going to make object detection bots from image and AI, but before that, if xs simulator can do the same for simulating the display for the camera , it will be great. Be ready to make the xs simulator camera so that you can make the necessary arrangements. Please let me know when you want to build the camera simulator

So that I can start the initial tests of object recognition from the image

**Reply by phoddie** [reply]
Thank you. I agree that simulator support will be important. We've done that kind of thing before. But.... the design of the Camera API needs to be driven by the embedded device (ESP32). So... we should understand how this will work on the embedded device first. That will then define how to integrate camera support into the simulator.

**Reply by phoddie** [reply]
I'm not a Telegram user. I could learn, but I very much prefer to keep technical conversations like this available to as many people as possible, so that they can learn and contribute too. Maybe our [Embedded JavaScript Gitter](https://app.gitter.im/#/room/#embedded-javascript_moddable:gitter.im) would be a good to try for conversations?
